src/main/java/org/jenkinsci/plugins/github_branch_source/GitHubSCMSource.java

```diff
@@ -128,6 +128,11 @@
      */
     // TODO remove once baseline Git plugin 3.0.2+
     private static final AtomicLong jenkins41244Warning = new AtomicLong();
+    /**
+     * How long to delay events received from GitHub in order to allow the API caches to sync.
+     */
+    private static /*mostly final*/ int eventDelaySeconds =
+            Math.min(300, Math.max(0, Integer.getInteger(GitHubSCMSource.class.getName() + ".eventDelaySeconds", 5)));
```

Comment:
I recommend writing

```java
// TODO 2.4+ use SystemProperties
```

so that it is easy to make a sweep through a plugin looking for stuff to refresh when bumping up the baseline.